substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/StackTraceUtils.java

```diff
@@ -155,3 +164,22 @@ public boolean visitFrame(FrameInfoQueryResult frameInfo) {
         }
     }
 }
+
+class GetClassContextVisitor extends JavaStackFrameVisitor {
+    private int skip;
+    final ArrayList<Class<?>> trace;
+
+    GetClassContextVisitor(final int skip) {
+        trace = new ArrayList<>();
+        this.skip = skip;
+    }
+
+    public boolean visitFrame(final FrameInfoQueryResult frameInfo) {
+        if (skip > 0) {
+            skip--;
+        } else if (StackTraceUtils.shouldShowFrame(frameInfo, true, false)) {
```

Comment:
@dmlloyd did you double-check on the HotSpot VM that `getClassContext` really is *not* filtering reflection methods (the `true` for the `showReflectFrames` parameter)? That would be a difference to `getCallerClass`.

Reply:
I believe it is not, but that is just from my memory.  I'll double-check it.

Reply:
Ah, it is in fact doing so.  I'll fix that.